tests/test_client.py

```diff
@@ -1391,3 +1396,71 @@ def test_read_metadata_remediation_message(client: GGClient):
     assert client.remediation_messages.pre_commit == messages["pre_commit"]
     assert client.remediation_messages.pre_push == messages["pre_push"]
     assert client.remediation_messages.pre_receive == messages["pre_receive"]
+
+
+LIST_MEMBERS_RESPONSE = [
+    {
+        "id": 3251,
+        "name": "Owl",
+        "email": "john.smith@example.org",
+        "role": "owner",
+        "access_level": "owner",
+        "active": True,
+        "created_at": "2022-04-20T11:07:24.000Z",
+        "last_login": "2022-04-20T11:07:24.000Z",
+    },
+    {
+        "id": 3252,
+        "name": "Owl",
+        "email": "john.smith@example.org",
+        "role": "owner",
+        "access_level": "owner",
+        "active": True,
+        "created_at": "2022-04-20T11:07:24.000Z",
+        "last_login": "2022-04-20T11:07:24.000Z",
+    },
+]
+
+
+@my_vcr.use_cassette("test_list_members.yaml", ignore_localhost=False)
```

Comment:
I am unsure why you think it would not work ? The `GITGUARDIAN_API_KEY` will require the related scopes (`members`, `teams` and `sources`) but it will work if those scopes are granted 🤔

Reply:
I gave a shot at writing tests that do not require mocking calls and rely only on cassettes, while feasible, it requires having few members and invitations sent in the workspace (also it will delete some resources so better do it on a test workspace), those are documented through comments on those tests. Let me know if you think it is fine the way it is now, I don't think mock would do great justice for the tests as you said :/